pkg/snmp/traps/config.go

```diff
@@ -46,14 +47,18 @@ type Config struct {
 // ReadConfig builds and returns configuration from Agent configuration.
 func ReadConfig(agentHostname string) (*Config, error) {
 	var c Config
-	err := config.Datadog.UnmarshalKey("snmp_traps_config", &c)
+	err := config.Datadog.UnmarshalKey("network_devices.snmp_traps", &c)
 	if err != nil {
 		return nil, err
 	}
 
+	if !c.Enabled {
+		return nil, errors.New("traps listener is disabled")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏This is not supposed to happen right ?

Reply:
In theory not at the moment, this function is not called if the traps listener is disabled. I'm ok to remove if you prefer